Mono.Debugging.Soft/PortablePdbData.cs

```diff
@@ -77,73 +80,30 @@ internal class SoftScope
 			public int LiveRangeEnd;
 		}
 
-		class JsonSourceLink
+		Stream GetStream()
 		{
-			[JsonProperty ("documents")]
-			public Dictionary<string, string> Maps { get; set; }
+			if (pdbBytes != null)
+				return new MemoryStream (pdbBytes);
+			return new FileStream (pdbFileName, FileMode.Open, FileAccess.Read, FileShare.Read);
 		}
 
-		class SourceLinkMap
+		public string GetSourceLinkBlob ()
 		{
-			public string RelativePathWildcard { get; }
-			public string UriWildcard { get; }
-
-			public SourceLinkMap (string relativePathWildcard, string uriWildcard)
-			{
-				UriWildcard = uriWildcard;
-				RelativePathWildcard = relativePathWildcard;
-			}
-		}
+			using (var stream = GetStream()) {
+				using (var provider = MetadataReaderProvider.FromPortablePdbStream (stream)) {
```

Comment:
By default, it closes the underlying stream, unless `LeaveOpen` is passed. Should we `pre-fetch metadata` though?

Reply:
> Should we pre-fetch metadata though?

This might fetch more data than we need. I think this option is for users who want to have full control over the stream after reading.